src/clj/harja/kyselyt/paikkaus.clj

```diff
@@ -213,9 +213,10 @@
   "Päivittää paikkauskohteen tilan harjan sisäisen id:n perusteella (lähetetty, virhe)."
   [db paikkauskohde]
   (let [id (::paikkaus/id paikkauskohde)
-        ehdot (if (id-olemassa? id)
-                {::paikkaus/id id})]
-    (update! db ::paikkaus/paikkauskohde paikkauskohde ehdot)
+        ehdot {::paikkaus/id id}]
+    (assert (id-olemassa? id) "Paikkauskohteen päivitys vaatii id:n")
```

Comment:
Täällä ei tykätä assertista: https://clojureverse.org/t/pragmatic-programmer-should-we-all-use-assert/9106/7

Here’s [Alex Miller on the topic 15](https://clojurians.slack.com/archives/C03S1KBA2/p1646924206922289). Emphasis mine because “logically impossible” is a key part of their intended use case:

assertions are designed to be turned off in production…they are designed to catch logically impossible situations that can only occur if the program is wrong. they should not be used to detect invalid input (like a bad web request, bad user input, etc).

Reply:
Joo, turha assertti. Käytännössä tuossa kohti on aina id, ja lisäksi nyt whenitän sen, eli updatehaaraan ei mennä ellei ole id:tä. Aiemmin siinä oli sellainen vika, että id:n puuttuessa olisi syntynyt update-lause ilman where-ehtoa, minkä halusin nyt eliminoida.